### PR TITLE
Follow-up fixes to CUB iterator test

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Those where suggested in a post-merge review by @gevtushenko.

### CODE REVIEW

File: cub/test/c2h/custom_type.cuh [hunk 1]
```diff
@@ -131,6 +131,11 @@ public:
   {
     return lhs.key == rhs.key && lhs.val == rhs.val;
   }
+
+  friend __host__ __device__ bool operator!=(const CustomType& lhs, const CustomType& rhs)
+  {
+    return !(lhs == rhs);
+  }
```
Comment (bernhardmgruber) [inline cub/test/c2h/custom_type.cuh:138]: Needed by `CompareDeviceResults`. It also makes sense for a `c2h::custom_type` which is `equal_comparable_t` to also support `!=`.

Comment (gevtushenko) [reply cub/test/c2h/custom_type.cuh:138]: `CompareDeviceResults` is legacy testing facility. Please, compare device vectors using `==` or `thrust::equal`.

File: cub/test/c2h/custom_type.cuh [hunk 2]
```diff
@@ -131,6 +131,11 @@ public:
   {
     return lhs.key == rhs.key && lhs.val == rhs.val;
   }
+
+  friend __host__ __device__ bool operator!=(const CustomType& lhs, const CustomType& rhs)
```
Comment (gevtushenko) [inline cub/test/c2h/custom_type.cuh:135]: suggestion: the motivation behind policy is to check if a given algorithm relies on documented set of type requirements. If we promise to only use `==` but accidentally rely on `!=` it'll be a problem. If you do need `!=`, maybe it's worth extracting into a separate policy?

**Review by gevtushenko — APPROVED** [review]

File: cub/test/c2h/custom_type.cuh (hunk 1 above)
Comment (bernhardmgruber) [reply cub/test/c2h/custom_type.cuh:138]: Even if this adds 2min of compile time to the iterator test?

File: cub/test/c2h/custom_type.cuh (hunk 2 above)
Comment (bernhardmgruber) [reply cub/test/c2h/custom_type.cuh:135]: If a type promises to be equality comparable, it should expose both, `==` and `!=`. This was even built into C++20, where the compiler can rewrite any expression `a != b` as `!(a == b)` if no `operator!=` is found. I strongly believe we should provide both operators for the policy. Also, any type that only defines `operator==` but not `operator!=` is broken.

File: cub/test/c2h/custom_type.cuh (hunk 1 above)
Comment (gevtushenko) [reply cub/test/c2h/custom_type.cuh:138]: > Even if this adds 2min of compile time to the iterator test?

Yes, I think so. I'd like to get rid of `CompareDeviceResults` as soon as possible. It allocates memory internally, without relying on c2h safe memory allocation we introduced for a reason. Besides that, the less we mask the problem by sticking to legacy solution, the sooner we'll prioritize fixing it.

**Review by gevtushenko — APPROVED** [review]

File: cub/test/c2h/custom_type.cuh (hunk 1 above)
Comment (bernhardmgruber) [reply cub/test/c2h/custom_type.cuh:138]: I see. I will change to comparing vectors than.

> the less we mask the problem by sticking to legacy solution, the sooner we'll prioritize fixing it.

You know, I kind of have a half-baked shot at that: #1870 :)

**Pull request merged**